### PR TITLE
feat(fiatAccounts): support PIX accounts

### PR DESCRIPTION
**Pull request by ailsoncgt**
[Pix](https://www.bcb.gov.br/estabilidadefinanceira/pix)

### CODE REVIEW

File: src/fiat-account.ts [hunk 1]
```diff
@@ -195,4 +218,4 @@ export type GetFiatAccountsResponse = z.infer<
 export const postFiatAccountResponseSchema = obfuscatedFiatAccountDataSchema
 export type PostFiatAccountResponse = z.infer<
   typeof postFiatAccountResponseSchema
->
+>
```
Comment (jophish) [inline src/fiat-account.ts:221]: nit: newline

**Review by jophish — COMMENTED** [review]
Thanks for the PR! Before this is approved and merged, please open a PR against the [FiatConnect specification document](https://github.com/fiatconnect/specification/blob/main/fiatconnect-api.md) itself including a description of these changes; that document is the original source of truth for all these other repositories.

**Comment by ailsoncgt** [comment]
@jophish can you please review [PR](https://github.com/fiatconnect/specification/pull/84) in specification .

**Comment by ailsoncgt** [comment]
@jophish my [PR](https://github.com/fiatconnect/specification/pull/84) in specification was merged. We can move forward here?

File: src/fiat-account.ts [hunk 2]
```diff
@@ -46,6 +57,18 @@ const requiredFiatAccountSchemaFieldsSchema = z.object({
...
+  z.object(
+    {
+      keyType: pixKeyTypeEnumSchema,
+      key: z.string(),
+      fiatAccountType: z.literal(FiatAccountType.BankAccount),
+    },
+    { description: 'pixAccountSchema' },
+  ),
+)
+export type PixKey = z.infer<typeof pixAccountSchema>
```
[the remark this answers is not in the document]
Comment (cajubelt) [reply src/fiat-account.ts:70]: and I think there's a typo here
```
export type PixKey = z.infer<typeof pixAccountSchema>
```
the type should be named `PIXAccount`

File: src/fiat-account.ts (hunk 1 above)
Comment (cajubelt) [reply src/fiat-account.ts:222]: still need a newline here

**Review by cajubelt — APPROVED** [review]

File: src/fiat-account.ts [hunk 3]
```diff
@@ -46,6 +57,50 @@ const requiredFiatAccountSchemaFieldsSchema = z.object({
   fiatAccountType: fiatAccountTypeSchema,
 })
 
+export const PIX_EMAIL_KEY_REGEX =
+  /^(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/ // credit to https://stackoverflow.com/a/201378/5807149
```
Comment (cajubelt) [inline src/fiat-account.ts:61]:
```suggestion
/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // credit to http://emailregex.com/
```

Comment (cajubelt) [reply src/fiat-account.ts:61]: sorry, my previous email regex suggestion failed a linter check for no control characters. This one looks simpler anyway

File: src/fiat-account.ts [hunk 4]
```diff
@@ -46,6 +57,50 @@ const requiredFiatAccountSchemaFieldsSchema = z.object({
...
+export const PIX_PHONE_KEY_REGEX = /^[0-9]{11}$/
+export const PIX_RANDOM_KEY_REGEX = /^[a-zA-Z0-9]{32}$/
+
+export const pixAccountSchema = requiredFiatAccountSchemaFieldsSchema
+  .and(
+    z.object(
+      {
+        fiatAccountType: z.literal(FiatAccountType.BankAccount),
+        keyType: pixKeyTypeEnumSchema,
+        key: z.string(),
```
Comment (cajubelt) [inline src/fiat-account.ts:72]:
```suggestion
```
My colleague pointed out to me that these are not useful, and might as well be removed for readability's sake

File: src/fiat-account.ts [hunk 5]
```diff
@@ -46,6 +57,48 @@ const requiredFiatAccountSchemaFieldsSchema = z.object({
   fiatAccountType: fiatAccountTypeSchema,
 })
 
+export const PIX_EMAIL_KEY_REGEX =
+/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // credit to http://emailregex.com/
```
Comment (cajubelt) [inline src/fiat-account.ts:61]:
```suggestion
  /* eslint-disable-next-line no-useless-escape */ // For some reason, eslint thinks the escaped \[ and /] are useless. they are indeed useful.
/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ // credit to http://emailregex.com/
```

**Review by cajubelt — APPROVED** [review]

**Comment by cajubelt** [comment]
Required formatting checks still fail, this time it's because you didn't run prettier. For the future, here are the list of checks to run on your local machine before asking for a review:
image

without passing these we can't merge (even if we'd like to, like this time). 

you can fix the prettier check with "yarn format"

**Comment by ailsoncgt** [comment]
ok @cajubelt , I will do that.

**Pull request merged**